src/odemis/dataio/tiff.py

```diff
@@ -1206,28 +1206,50 @@ def _findImageGroups(das):
     current_ifd = 0
     prev_da = None
 
+    # required keys for merging
+    required_keys = [model.MD_IN_WL, model.MD_OUT_WL]
+    # metadata -> default value if missing
+    matching_keys = {
+        model.MD_HW_NAME: None,
+        model.MD_HW_VERSION: None,
+        model.MD_PIXEL_SIZE: None,
+        model.MD_POS: None,
+        model.MD_ROTATION: 0,
+        model.MD_SHEAR: 0,
+    }
     for da in das:
         # check if it can be part of the current group (compare just to the previous DA)
         dims = da.metadata.get(model.MD_DIMS, "CTZYX"[-da.ndim::])
-        if (prev_da is None
-            or da.shape[0] != 1  # If C != 1 => not possible to merge (C is always first dimension) or it is a
-                # greyscale YX image
-            or (model.MD_IN_WL not in da.metadata or model.MD_OUT_WL not in da.metadata)
-            or prev_da.shape != da.shape
-            or prev_da.metadata.get(model.MD_HW_NAME, None) != da.metadata.get(model.MD_HW_NAME, None)
-            or prev_da.metadata.get(model.MD_HW_VERSION, None) != da.metadata.get(model.MD_HW_VERSION, None)
-            or prev_da.metadata.get(model.MD_PIXEL_SIZE) != da.metadata.get(model.MD_PIXEL_SIZE)
-            or prev_da.metadata.get(model.MD_POS) != da.metadata.get(model.MD_POS)
-            or prev_da.metadata.get(model.MD_ROTATION, 0) != da.metadata.get(model.MD_ROTATION, 0)
-            or prev_da.metadata.get(model.MD_SHEAR, 0) != da.metadata.get(model.MD_SHEAR, 0)
-           ):
-            # new group
+
+        plen_c, len_c = 1, 1
```

Comment:
You shouldn't reset `plen_c` to 1 every time.
Just set it to `None` before the `for` loop, like `prev_da`.